src/report/report-system/account.scm

```diff
@@ -0,0 +1,73 @@
+(define-module (gnucash report report-system account))
+(use-modules (gnucash gnc-module))
+(use-modules (gnucash gnc-module))
+(use-modules (ice-9 hash-table))
+
+(use-modules (srfi srfi-1)
+             (srfi srfi-13))
+
+(cond-expand
+  (guile-2
+    (eval-when
+      (compile load eval expand)
+      (gnc:module-load "gnucash/engine" 0)))
+  (else
+    (gnc:module-load "gnucash/engine" 0)))
+
+(export account-same?)
+(export account-in-list?)
+(export account-in-list-pred)
+(export account-in-alist)
+(export account-full-name<?)
+(export account-list-predicate)
+(export accounts-get-children-depth)
+
+;; is account in list of accounts?
+(define (account-same? a1 a2)
+  (string=? (gncAccountGetGUID a1) (gncAccountGetGUID a2)))
+
+(define account-in-list?
```

Comment:
Since you're already using SRFI-1, perhaps this could be just `(not (equal? #f (find (lambda (acc) (account-same? account acc)) accounts)))`?